modules/jooby-avaje-inject/src/test/java/io/jooby/avaje/inject/app/Controller.java

```diff
@@ -0,0 +1,36 @@
+package io.jooby.avaje.inject.app;
+
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.typesafe.config.Config;
+import io.avaje.inject.InjectModule;
+import io.jooby.annotation.GET;
+import io.jooby.annotation.Path;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+@Singleton
+@Path("")
+@InjectModule(requires = {JsonMapper.class, Config.class, String.class})
```

Comment:
never use Avaje Inject. Why do we need this line?

Reply:
```text
In Avaje Inject, the dependency graph is typically validated when the application compiles. As beans provided by Jooby Modules are registered at runtime, you must add @InjectModules(requires={String.class, <other Jooby Module Provided Bean ClassNames>…​}) to inform the avaje processor that these beans are provided at runtime.
```
from https://jooby.io/#dependency-injection-avaje-inject-mvc-routes :)